backend/canisters/user_index/impl/src/queries/referral_metrics.rs

```diff
@@ -0,0 +1,81 @@
+use crate::{read_state, RuntimeState};
+use ic_cdk_macros::query;
+use std::{cmp::Reverse, collections::HashMap};
+use types::UserId;
+use user_index_canister::referral_metrics::{Response::*, *};
+
+#[query]
+fn referral_metrics(_args: Args) -> Response {
+    read_state(referral_metrics_impl)
+}
+
+#[derive(Default)]
+struct ReferralData {
+    paid_diamond: u32,
+    unpaid_diamond: u32,
+    icp_raised_for_paid_diamond: u32,
+}
+
+fn referral_metrics_impl(runtime_state: &RuntimeState) -> Response {
+    let mut user_referrals_map: HashMap<UserId, ReferralData> = HashMap::new();
+
+    for user in runtime_state.data.users.iter() {
+        if let Some(referred_by) = user.referred_by {
+            let data = user_referrals_map.entry(referred_by).or_default();
+            let icp_raised_for_paid_diamond: u64 =
+                user.diamond_membership_details.payments().iter().map(|p| p.amount_e8s).sum();
+            if icp_raised_for_paid_diamond > 0 {
+                data.paid_diamond += 1;
+                data.icp_raised_for_paid_diamond += (icp_raised_for_paid_diamond / 100_000_000) as u32;
```

Comment:
This is going to do integer division, so all values < 1 will be 0. So 1 month and 3 month payments will be 0.

Reply:
I think you should just leave everything as e8s, since the payments are as low as 0.2 ICP.